demo/src/main/java/com/novoda/storagepathfinder/demo/StoragePathsAdapter.kt

```diff
@@ -0,0 +1,34 @@
+package com.novoda.storagepathfinder.demo
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.novoda.storagepathfinder.StoragePath
+
+typealias Listener<T> = (T) -> Unit
+
+internal class StoragePathsAdapter(private val layoutInflater: LayoutInflater,
+                                   private val deviceStoragePaths: List<StoragePath>,
+                                   private val onAddFileClicked: Listener<StoragePath>) : RecyclerView.Adapter<StoragePathViewHolder>() {
+
+    init {
+        setHasStableIds(true)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoragePathViewHolder {
+        return StoragePathViewHolder(layoutInflater.inflate(R.layout.item_device_storage_root, parent, false))
+    }
+
+    override fun onBindViewHolder(holder: StoragePathViewHolder, position: Int) {
+        holder.bind(deviceStoragePaths[position], onAddFileClicked)
+    }
+
+    override fun getItemCount(): Int {
```

Comment:
this and the next one can be inlined:
`override fun getItemCount(): Int = deviceStoragePaths.size`

Reply:
I mean I could but this is an override, I'm forced to implement these 😄

Reply:
You still can!
Like this example, that is valid 
```
open class A {
    open fun getItemCount(): Int {
        return 0
    }
}

class B : A() {
    override fun getItemCount() = 2
}
```

Reply:
Oooooo 💥